parquet/src/main/java/org/apache/iceberg/data/parquet/BaseParquetWriter.java

```diff
@@ -50,6 +54,32 @@ protected ParquetValueWriter<T> createWriter(MessageType type) {
   protected abstract ParquetValueWriters.StructWriter<T> createStructWriter(
       List<ParquetValueWriter<?>> writers);
 
+  protected ParquetValueWriter<?> fixedWriter(ColumnDescriptor desc) {
+    return new FixedWriter(desc);
+  }
+
+  protected ParquetValueWriters.PrimitiveWriter<?> dateWriter(ColumnDescriptor desc) {
+    return new DateWriter(desc);
+  }
+
+  protected ParquetValueWriters.PrimitiveWriter<?> timeWriter(ColumnDescriptor desc) {
+    return new TimeWriter(desc);
+  }
+
+  protected ParquetValueWriters.PrimitiveWriter<?> timestampWriter(
+      ColumnDescriptor desc, boolean isAdjustedToUTC) {
+    if (isAdjustedToUTC) {
+      return new TimestamptzWriter(desc);
+    } else {
+      return new TimestampWriter(desc);
+    }
+  }
+
+  protected ParquetValueWriters.PrimitiveWriter<?> uuidWriter(ColumnDescriptor desc) {
+    // Use primitive-type writer; no special writer needed.
```

Comment:
There is no primitive writer for UUIDs. Maybe this was a copy/paste error?

Reply:
It uses the fixed length primitive writer with byte[] as input.

Reply:
That doesn't work because the UUID representation is `UUID`.

Reply:
All the generic reader writer test case uses UUID as byte[], hence it is working. 

You mean existing testcases are wrong?
https://github.com/apache/iceberg/blob/be6e9daf8901cdee63197e77fcf95624bb694f39/arrow/src/test/java/org/apache/iceberg/arrow/vectorized/ArrowReaderTest.java#L861-L862

Reply:
hmm,
Only Arrow has the testcase for UUID, other usage of generic writers doesn't cover UUID data type!

Reply:
Okay, I did some digging and it looks like this behavior was accidentally introduced when moving to the type annotation visitor (https://github.com/apache/iceberg/commit/1d13a10d4c8a1174083eb81373310b8499f84f44). Before that point, UUID was not supported but after that it would fall through to the primitive type and return a writer for bytes.

The problem is that this conflicts with the representation used by the Avro generic object model: https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/avro/ValueReaders.java#L108-L110

I think this change is the right choice because the object models should match and using UUID for this is well-established in the Avro implementation and the generic random data generator.